client/src/routes/App/FileTransfer/FileTransfer.js

```diff
@@ -292,17 +292,17 @@ class FileTransfer extends PureComponent {
           isSelectorEnabled: false,
         });
       },
-      onDone: (file, meta) => {
+      onDone: (files) => {
         if (file !== undefined) {
           if (Array.isArray(file)) {
-            file.forEach(file => {
-              file.getBlob((err, blob) => download(blob, file.name));
-            });
+            multiDownload(
+              // make regular File from webtorrent File https://github.com/webtorrent/webtorrent/blob/v2.4.14/lib/file.js#L7
+              files.map(file => new File([file.getBlob()], file.name, {type: file.type}))
```

Comment:
btw, it looks like [`getBlob`](https://github.com/webtorrent/webtorrent/blob/v1.9.7/lib/file.js#L123) was removed in webtorrent v2.

this seems related to them switching to local storage in v2 as opposed to keeping complete torrents in RAM in v1 ref https://github.com/webtorrent/webtorrent/issues/86#issuecomment-419445849

Reply:
Blaze currently uses Webtorrent v1, and we'll need to upgrade to v2. From what I can recall since the last time I had checked, there were a bunch of API changes associated with v2 that we'd also have to make in Blaze which I wasn't fond of picking up then. Would you be interested in picking this up?

Reply:
yeah, I'm also afraid it's a can of worms... the local storage feature might make it worthwhile though. I guess a 10GB file would now fill up 10GB of my swap right?

Reply:
judging by the File [API docs](https://github.com/webtorrent/webtorrent/blob/v2.4.14/docs/api.md#file-api), I think using streamURL blaze can initiate download before the torrent has finished downloading, and so download.js could theoretically be called upon as soon as the torrent is initiated

Reply:
file.getBlob and file.blob are functionally identical, it was simply renamed to match the W3C file API, rather than create our own structure, but getBlob also loaded the entire file into memory

Reply:
yep, so the transition to v2 shouldn't be too big of a hassle (API wise) when keeping the 'whole file in memory' pattern

Reply:
ah just saw your cross [comment](https://github.com/webtorrent/webtorrent/issues/2849#issuecomment-2266669754), so `file.streamURL` is a viable option to get around the 'whole file in memory' pattern in the current downloading setup, with minimal overhead from `client.createServer`. that's great to hear and probably the way to move forward there!

Reply:
> the local storage feature might make it worthwhile though

@ddelange isn't the local storage limited to like 10MBs? I wonder how it would manage files of much higher sizes in such a limited space

Reply:
in v2 the torrents directly download to [`/tmp/webtorrent`](https://github.com/webtorrent/webtorrent/blob/v2.4.14/docs/api.md?plain=1#L120) by default using file system access (FSA): https://github.com/ThaUnknown/fsa-chunk-store

`path` can be set on a pee-torrent basis by the receiver before the torrent is added (folder selection is google chrome only if I understood the comments correctly)

Reply:
> in v2 the torrents directly download to [/tmp/webtorrent](https://github.com/webtorrent/webtorrent/blob/v2.4.14/docs/api.md?plain=1#L120) by default using [file system access](https://github.com/ThaUnknown/fsa-chunk-store).

for the sake of simplicity lets say sure, but if you want specifics, it actually writes to individual chunks via FSA's origin private file system, which functionally is no different from the likes of localStorage or IDB, except its a LOT faster, and allows for MUCH, MUCH, MUCH bigger file sizes, in extreme cases up to terabytes

on chrome that's a bit different, because IF YOU CHOOSE TO specify a rootDir, it can emulate the file structure in a specified folder as already mentioned, on top of saving chunks as cache to OPFS

Reply:
thanks for pointing to rootDir!

api.md [points to](https://github.com/webtorrent/webtorrent/blob/v2.4.14/docs/api.md?plain=1#L155) storeOpts.rootDir but I think that [should be](https://github.com/webtorrent/webtorrent/blob/v2.4.14/lib/torrent.js#L535) opts.rootDir right?

based on the [comments](https://github.com/webtorrent/webtorrent/blob/v2.4.14/docs/api.md?plain=1#L120) in api.md, I'm now not sure what's the difference between opts.path and opts.rootDir. should I ignore opts.path and only work with opts.rootDir?